app.js

```diff
@@ -55,14 +56,15 @@ app.use(function(req, resp, next){
 app.use(reactRouted);
 app.use(express.static(__dirname + '/public', {maxAge: 3600000}));
 app.use(function(req, res, next) {
-  var firstPath = req.url.split("/")[1];
-  var search = req.search || "";
+  var location = url.parse(req.url).pathname;
```

Comment:
Oddly I was finding that req.url had the search in it still, so I decided to do an explicit parse for search and pathname, and use the same tech to do both to ensure consistency.